src/api/queries/camps.ts

```diff
@@ -0,0 +1,98 @@
+import { computed } from 'vue'
+import { useMutation, useQuery, useQueryClient } from '@tanstack/vue-query'
+import { useRoute } from 'vue-router'
+import { apiClient } from '@/api/apiClient'
+import type { components } from '@/api/schema'
+
+type Camp = components['schemas']['CampResponse']
+type CampRequest = components['schemas']['CampRequest']
+
+const fetchCamps = async (): Promise<Camp[]> => {
+  const { data, error } = await apiClient.GET('/api/camps')
+  if (error) throw error
+  return data
+}
+
+const createCamp = async (variables: { newCamp: CampRequest }): Promise<Camp> => {
+  const { newCamp } = variables
+  const { data, error } = await apiClient.POST('/api/admin/camps', {
+    body: newCamp,
+  })
+  if (error) throw error
+  return data
+}
+
+const updateCamp = async (variables: {
+  campId: number
+  updatedCamp: CampRequest
+}): Promise<Camp> => {
+  const { campId, updatedCamp } = variables
+  const { data, error } = await apiClient.PUT('/api/admin/camps/{campId}', {
+    params: { path: { campId } },
+    body: updatedCamp,
+  })
+  if (error) throw error
+  return data
+}
+
+const deleteCamp = async (variables: { campId: number }): Promise<void> => {
+  const { campId } = variables
+  const { error } = await apiClient.DELETE('/api/admin/camps/{campId}', {
+    params: { path: { campId } },
+  })
+  if (error) throw error
+}
+
+export const useCampsQuery = () => {
+  return useQuery({
+    queryKey: ['camps'],
+    queryFn: fetchCamps,
+  })
+}
+
+export const useCurrentCampQuery = () => {
+  const route = useRoute()
+  const campname = computed(() => route.params.campname as string)
+  return useQuery({
+    queryKey: ['camps'],
+    queryFn: fetchCamps,
+    select: (camps) => camps.find((camp) => camp.displayId === campname.value),
+    enabled: computed(() => !!campname.value),
+  })
+}
+
+export const useCreateCampMutation = () => {
+  const queryClient = useQueryClient()
+  return useMutation({
+    mutationFn: createCamp,
+    onSuccess: (data) => {
+      queryClient.setQueryData(['camps'], (old: Camp[] | undefined) =>
```

Comment:
型引数を使うとより短い記述で`old`が`Camp[] | undefined`に推論されると思います（他の`setQueryData`使用箇所も同様）
```suggestion
      queryClient.setQueryData<Camp[]>(['camps'], (old) =>
```